lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
+
+      // Match category of URN
+      switch (cat) {
+        case 'pbk:ec:secp256r1':
+          // Check for compressed/uncompressed ECDSA public keys
+          if (charArr[0] === '02' || charArr[0] === '03') {
+            // Check length of compressed public key
+            if (charArr.length !== 25) {
+              return false;
+            }
+          } else if (charArr[0] === '04') {
+            // Check length of uncompressed public key
+            if (charArr.length !== 65) {
+              return false;
+            }
+          }
+
+          break;
+        case 'pbk:rsa:2048':
+          // Grab RSA exponent
+          var exponent = id.substring(id.length - 6);
+
+          // Check if RSA exponent is in binary format or if the total ID length is proper size (512 bytes + 6 bits)
+          if (!exponent.matches("[01]+") || id.length !== 518) {
```

Comment:
Exponent is actually not in a binary format, it's hex (https://github.com/chronicled/open-registry-ethereum#rsa-public-key). 
Also idea here is that exponent may or may not be provided. If it's not provided than we use default one as here https://github.com/chronicled/open-registry-utils/blob/master/lib/rsa.js#L25
Also length is not 512 bytes, but 256 bytes (2048 / 8), it's 512 hex symbols (nibbles). 1 byte = 2 nibbles
So first we should check if id is equal / greater than 256 bytes and then if exponent is provided, check if it's of even size in hex symbols.